### ISSUE TITLE
VisualStateManager does not seem to work on Android

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#5449]
### Description

Using a VisualStateManager to highlight selected items in a CollectionView works fine in Windows but not in Android. The sample one I'm using here is:
```
<VisualStateManager.VisualStateGroups>
    <VisualStateGroup Name="CommonStates">
        <VisualState Name="Normal" />
        <VisualState Name="Selected">
            <VisualState.Setters>
                <Setter Property="BackgroundColor" Value="Yellow" />
            </VisualState.Setters>
        </VisualState>
    </VisualStateGroup>
</VisualStateManager.VisualStateGroups>
```

This should highlight any selected items (if there are any) by drawing their background in yellow.

### Steps to Reproduce

1. Unzip [MAUIListEntry-VisualState.zip](https://github.com/dotnet/maui/files/8310551/MAUIListEntry-VisualState.zip)
2. Compile and run it on Windows
3. Click on one of the list items, note its background changes to yellow
4. click on the other one, note that its background changes to yellow and the one that was formerly yellow goes back to white
5. Recompile and run the program on Android
6. Repeat actions 3 & 4 and observe that the list item background does not change.
 

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11, Windows 10 

### Did you find any workaround?

You can code around it for single selections using the SelectionChanged event to explicitly handle changing background color but I've no idea how general that solution is.

**Comment by Amir-Care** [comment]
Verified repro with Version 17.2.0 Preview 2.0 [32208.484.main]
Sample project: [#5449.zip](https://github.com/dotnet/maui/files/8313399/5449.zip)

**Comment by antonfirsov** [comment]
There are two issues here:
1. `VisualStateManager.GoToState` is not hooked into the Android adapter code. This is easy to fix. However:
2. "Switching back" to unselected state seems to be implemented by setting the property (in this case `BackgroundColor`) to `null` which doesn't do anything. This is essentially identical to #5973 which is a design decision. Can anyone suggest how to address this?

This is where `BackgroundColor`'s `UnApply` lands:
https://github.com/dotnet/maui/blob/de0952445724b15f39ef8b4cff24ea70cecc9d43/src/Controls/src/Core/Setter.cs#L120-L121

Then in `ClearValue` it will set `newValue == null`:
https://github.com/dotnet/maui/blob/de0952445724b15f39ef8b4cff24ea70cecc9d43/src/Controls/src/Core/BindableObject.cs#L85

**Comment by antonfirsov** [comment]
@david-maw this is essentially fixed with #6249, but note that you have to define a default `BackgroundColor` for your grid, otherwise VisualStateManager won't be able to determine what color to switch back to after deselection:


https://github.com/dotnet/maui/blob/e79f1331d094ea4e1b7b99b60c2f10aa43aed838/src/Controls/samples/Controls.Sample/Pages/Controls/CollectionViewGalleries/SelectionGalleries/VisualStatesGallery.xaml#L15-L32

This is by design, see #6183 and #1485 for more info.

**Comment by david-maw** [comment]
Thanks for the fix @antonfirsov I can code around that constraint but it's a pain, there are a few consequences of that design decision that make migration from Xamarin harder, so I've raised Issue #6454 to track them. On the face of it that's an awkward design decision to detect and handle in migrated code, but there's undoubtedly more to it than I know.
